components/places/tests/fennec_bookmarks.rs

```diff
@@ -140,6 +140,21 @@ fn test_import_unsupported_db_version() -> Result<()> {
 
 #[test]
 fn test_import() -> Result<()> {
+    use places::api::places_api::ConnectionType;
+    use url::Url;
+
+    fn bookmark_exists(places_api: &PlacesApi, url_str: &str) -> Result<bool> {
+        let url = Url::parse(url_str)?;
+        let conn = places_api.open_connection(ConnectionType::ReadOnly)?;
+        Ok(conn.query_row_and_then(
+            "SELECT COUNT(*) from main.moz_bookmarks b
```

Comment:
Nit: `FROM` should be capitalized.

Also, this is test code, so doesn't really matter, but `SELECT EXISTS` (a la https://github.com/mozilla/application-services/blob/master/components/logins/src/db.rs#L582) would probably be better.